### ISSUE TITLE
Crash in group_add_addrlist

### ISSUE THREAD

**Issue by megous** [neomutt/neomutt#1885]
Looks like there's a missing initialization of `g->al` queue when creating a group (that's the obvious part):

```
@@ -74,8 +74,9 @@ struct Group *mutt_pattern_group(const char *pat)
     mutt_debug(LL_DEBUG2, "Creating group %s\n", pat);
     g = mutt_mem_calloc(1, sizeof(struct Group));
     g->name = mutt_str_strdup(pat);
     STAILQ_INIT(&g->rs);
+    TAILQ_INIT(&g->al);
     mutt_hash_insert(Groups, g->name, g);
   }
 
   return g;
```

This fixes the crash for me, but I also have to remove the assertion check to make neomutt not crash on startup.

```
@@ -187,9 +188,9 @@ static void group_add_addrlist(struct Group *g, const struct AddressList *al)
   TAILQ_FOREACH_SAFE(a, &al_new, entries, tmp)
   {
     mutt_addrlist_append(&g->al, a);
   }
-  assert(TAILQ_EMPTY(&al_new));
+//  assert(TAILQ_EMPTY(&al_new));
 }
 
 /**
  * group_add_regex - Add a Regex to a Group
```

I can't wrap my head around why the `al_new` does not end up empty after the previous loop. It looks like it's initialized properly.

The crash happens on the first group command in my .muttrc:

```
group -group gname -addr addr@dom
```

**Comment by megous** [comment]
Reproducer:

```
echo "group -group x -addr asdf@sdf.cz" > t
neomutt -F t
```

**Comment by megous** [comment]
Hmm, looks like a dup of #1882 but why the assert needs to be removed?

**Comment by megous** [comment]
Ah I see, `TAILQ_INSERT_TAIL` in `mutt_addrlist_prepend` is not enough to properly remove elements properly from `al_new`.

**Comment by flatcap** [comment]
Hi @megous 
Thanks for the bug report and thanks for taking the time to do some analysis / bug-fixing.
The fix has been merged in f9b997d02.
